### PR TITLE
Add coercion support for loopback data type: geopoint

### PR DESCRIPTION
**Pull request by gunjpan**
Connect to #365 

**Update 17/11/16**:
  - Blocked until new geopoint module is published
**Update 18/11/16**:
 - Updated implementation to use new module, will rebase once new module is published on npm

### CODE REVIEW

**Comment by gunjpan** [comment]
Update: After discussing with @bajtos, we have decided to extract out `geo` class from [loopback-datasource-juggler](https://github.com/strongloop/loopback-datasource-juggler/blob/72c27b040c61e217d229b792e4db54275a034dec/lib/geo.js)  as an independent module and use it here for dealing with `GeoPoint` object.

File: lib/types/geopoint.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    err.statusCode = 400;
+    return err;
+  }
+}
+
+function coerceObject(value) {
+  var result = {};
+  for (var key in value) {
+    var val = value[key];
+    result[key] = Number(val);
```
Comment (bajtos) [inline lib/types/geopoint.js]: Please use number converter's `fromSloppyValue` to parse numbers from a string.

File: test/rest-coercion/jsonform-geopoint.suite.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  });
+
+  describe('json form - geopoint - optional', function() {
+    // See verifyTestCases' jsdoc for details about the format of test cases.
+    verifyTestCases({ arg: 'arg', type: 'geopoint' }, [
+      // Empty values
+      [EMPTY_BODY, undefined],
+      // [{ arg: null }, null],
+      // [{ arg: {}}, {}],
+      // [{ arg: [] }, {}],
```
Comment (bajtos) [inline test/rest-coercion/jsonform-geopoint.suite.js]: Why are these three lines commented out? Either un-comment and keep them, or remove them completely.

File: test/rest-coercion/urlencoded-geopoint.suite.js [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+function suite(prefix, ctx) {
+  var EMPTY_QUERY = ctx.EMPTY_QUERY;
+  var ERROR_BAD_REQUEST = ctx.ERROR_BAD_REQUEST;
+  var verifyTestCases = ctx.verifyTestCases;
+
+  describe(prefix + ' - geopoint - required', function() {
+    // See verifyTestCases' jsdoc for details about the format of test cases.
+    verifyTestCases({ arg: 'arg', type: 'geopoint', required: true }, [
+      // Valid values - nested keys
+      ['arg[lat]=2.5&arg[lng]=3', new GeoPoint(2.5, 3)],
```
Comment (bajtos) [inline test/rest-coercion/urlencoded-geopoint.suite.js]: Please add a test for `arg[0]=2.5&arg[1]=3` too.

File: test/rest-coercion/urlencoded-geopoint.suite.js [hunk 4]
```diff
@@ -22,7 +22,8 @@ function suite(prefix, ctx) {
     // See verifyTestCases' jsdoc for details about the format of test cases.
     verifyTestCases({ arg: 'arg', type: 'geopoint', required: true }, [
       // Valid values - nested keys
-      ['arg[lat]=2.5&arg[lng]=3', new GeoPoint(2.5, 3)],
+        ['arg[lat]=2.5&arg[lng]=3', new GeoPoint(2.5, 3)],
+        ['arg[0]=2.5&arg[1]=3', new GeoPoint(2.5, 3)],
```
Comment (bajtos) [inline test/rest-coercion/urlencoded-geopoint.suite.js]: I think the indentation is off here, PTAL.

**Comment by gunjpan** [comment]
@slnode test please

**Pull request merged**